collector.go

```diff
@@ -50,10 +56,64 @@ func NewCollector(config CollectorConfig) (*Collector, error) {
 }
 
 func (c *Collector) Bundles() []Bundle {
-	return c.bundles
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	return CopyBundles(c.bundles)
```

Comment:
Why do we return a copy?

Reply:
Because a slice is a pointer and callers manipulating the received bundles could manipulate them for other callers as well. It is a safety net.